### ISSUE TITLE
BUG: Collations for projection tables are not deterministic

### ISSUE THREAD

**Issue by kitsunet** [neos/neos-development-collection#4729]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Projection tables use some system dependent charset/collation.

### Expected Behavior

Projection tables should use configured default charset/collation

### Steps To Reproduce

cr:setup on different systems results in different charsets and collations.

### Environment

```markdown
9.0
```

**Comment by kitsunet** [comment]
The strange collation you found (this is copied over from #4725 ) is probably a result of some defaults of your system, MY projections use utf8_unicode_ci (from the docs: "An alias for utf8mb3. In MySQL 8.0, this alias is deprecated; use utf8mb4 instead. utf8 is expected in a future release to become an alias for utf8mb4." )

I started to investigate a bit and the events table does use "proper" utf8mb4_unicode_ci and I think it's this here:
https://github.com/neos/eventstore-doctrineadapter/blob/main/src/DoctrineEventStore.php#L165

which seems missing from the schema construction we do for the contentrepository projections and I think we should definitely add it otherwise the charset and collaction are probably left to some system default which would be a bad idea.

See eg. here:
https://github.com/neos/contentgraph-doctrinedbaladapter/blob/9.0/src/DoctrineDbalContentGraphSchemaBuilder.php

Flow does set:

        defaultTableOptions:
          charset: 'utf8mb4'

So that seems to fit. IMHO we should first change all projections to actually adhere to this and then also use this to decide which collation to use in queries like these here.

Edit to make this clearer, my local system also has the projection tables on utf8mb3_unicode_ci the ones above were from my localbeach instances, so it seems indeed to depend on some default or other. Not good.
Tried to create a table with a varchar field in both local and localbeach and both came out different but neither of them with mb3 charset. This is so tricky, but I guess using the same configured default as we do for the eventstore would make a lot of sense.
